### PR TITLE
Todo list

### PR DESCRIPTION
**Pull request by vidhishweta01**
Todo list is an app which create list of projects along with their date. we can also add task to each project by default one project "today" is given to which any task can be added, task name cannot be empty. here local storage has been used to save data.
there is a button clear all it will reset the project entirely 
live link:
https://vidhishweta01.github.io/todo_list/

### CODE REVIEW

**Review by MkrtichSargsyan — CHANGES_REQUESTED** [review]
## _Status:_ Changes Requested :recycle:

## Hello Shweta Srivastava 


## To Highlight :tada:

  1. Well structured files
  2. It uses localStorage to save todos between sessions


However, there are some changes that you need to make in order to get your project approved

- [ ] The project is not desciptive enough in the readme file. Kindly add more description
- [ ] In the readme file change ```cd restaurant_page``` to ```cd todo_list```
- [ ] Kindly do not write commit messages with the same name
- [ ] Remove the commented imports from index.js file
- [ ] There should be a 'default' project
- [ ] The user can assign a to-do list items to a project
- [ ] User can view all to-dos in each project
- [ ] User can edit the details of a to-do item
- [ ] User can delete a to-do

- [ ] Necessary run common commands like build, development should be in package.json inside `scripts'. Here is a basic example [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/scripts_requirement.png)

      "scripts": {
        "dev": "webpack --mode development",
        "build": "webpack --mode production",
        "start": "webpack-dev-server --mode development --open" // optional
      },

- [ ] Kindly use arrow functions instead of the function keyword [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png)

<hr>

## Optional

- [ ] the forms used to create a project or a todo should have validations, you should not be able to create an empty project/todo 

Please, make the REQUIRED changes and submit your request again! :grinning:
Good luck. 

> ps: . If you have any questions about my review or need help, you can message me on slack ( @mkrtichsargsyan )

File: README.md [hunk 1]
```diff
@@ -1 +1,48 @@
...
+
+Download or clone this repo by using :
+
+## $ git clone https://github.com/vidhishweta01/todo_list.git
+
+do follow these steps given below:
+
+1. Change directory to the root of project.
+
+## $ cd restaurant_page
```
Comment (MkrtichSargsyan) [inline README.md]: change ```cd restaurant_page``` to ```cd todo_list```

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import setup from './js/dom';
+import './style.css';
+import Project from './js/project';
+import { updateStorage, GetData, store } from './js/storage';
+// import { GetData } from './js/storage';
+// import { store } from './js/storage';
```
Comment (MkrtichSargsyan) [inline src/index.js]: Kindly remove the commented imports

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, thank you for working on changes suggested previously :ok_hand:
## To Highlight 🎉
- [ ] Git flow is used properly.
- [ ] README contains instructions to set the project up locally.
## To Improve 🙏
- [x] There should be a 'default' project in the app. Please add a default project so that the user can assign items to it if he does not want to create a new project.
- [x] User should be able to edit a task. Please add this functionality to to-do items.
- [x] User should be able to delete a specific item. Please add this functionality.
- [x] Please add commands like build, and development to scripts. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/scripts_requirement.png).
- [x] You are still using function keywords in some modules like `storage.js` and `index.js`. Please use ES6 arrow functions instead of function keywords. Kindly see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/es6_requirement.png).
- [x] All necessary packages should be added in package.json like `webpack`, `webpack-cli`, `webpack-dev-server` etc. Also, those packages should be added as devDependencies. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/devdep_requirement.png). 
### Happy Coding ☺️
## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by vidhishweta01** [comment]
there is a default project named today where you can add your task

in developer dependencies webpack cli and webpack is added

 'this' keyword  is undefined in arrow function only then I have not used arrow function.

Screenshot6

Screenshot from 2021-07-08 12-56-23

**Comment by vidhishweta01** [comment]
please go through readme as it has every detail which was asked in code review
live link
https://vidhishweta01.github.io/todo_list/

File: src/index.js [hunk 3]
```diff
@@ -123,7 +123,7 @@ function showTasks() {
   free(div);
   const c = this.children;
   let txt = '';
-  for (let i = 0; i < c.length; i = i + 1) {
+  for (let i = 0; i < c.length; i = i + 1) { // eslint-disable-line
```
Comment (afanimartin) [inline src/index.js:126]: - Use ES6 syntax throughout this module. You still have functions declared with the `function` keyword. Functions like `addTask`, `deleteProject`, etc
- Do the same with functions in `storage.js` module.

**Review by afanimartin — CHANGES_REQUESTED** [review]
## Changes Required 🙏🏿 🙏🏿 🙏🏿  
Nice work making the requested change(s). Let's fix a few things before merging.

### Change(s) 🔕 🔕 🔕 
- Add `webpack-dev-server` to `devDependencies` section as requested
- Use ES6 syntax for functions in `index.js` and `storage.js` modules.


### Martin Afani
#### Slack: whiz25

**Comment by vidhishweta01** [comment]
arrow function has been used throughout the project
## I have used 'function' keyword only where I have used "this" keyword because it is undefined in arrow function    

In the project there is not mentioned specifically that I cannot use 'function" keyword. 

Is it depricated to use some traditional feature in the project??
https://www.theodinproject.com/paths/full-stack-javascript/courses/javascript/lessons/todo-list

https://www.tutorialspoint.com/es6/es6_functions.htm
## Arrow convention is causing error, this method is also given in E6 syntax 

Screenshot from 2021-06-27 23-35-42

Screenshot from 2021-06-27 23-42-03

**Review by afanimartin — APPROVED** [review]
## Approved 👍🏿 👍🏿 👍🏿 
Nice work making the requested changes and explaining why you wouldn't use the ES6 syntax to define some functions. Merge and proceed to the next project.


### Martin Afani
#### Slack: whiz25

**Pull request merged**